src/Microsoft.DotNet.Cli.Utils/StreamForwarder.cs

```diff
@@ -83,11 +83,6 @@ public void Read(TextReader reader)
 
         private void WriteBuilder()
         {
-            if (_builder.Length == 0)
```

Comment:
Won't you get multiple `\r` characters this way?

On Windows, the input will have `\r\n`. The Read method above will write the `\r` to the builder. Then when it reads the next character, it will call WriteLine when it sees `\n`.  Next the WriteLine method appends a Environment.NewLine to the end of the string, which then results in `\r\r\n`.

Doing it this way seems incorrect.  Instead I would imagine the Read method looking something like:

```C#
             while ((readCharacterCount = reader.Read(buffer, 0, bufferSize)) > 0)
             {
                 currentCharacter = buffer[0];
 
                 if (! s_ignoreCharacters.Contains(currentCharacter))
                 {
                     _builder.Append(currentCharacter);

                    if (currentCharacter == s_flushBuilderCharacter)
                    {
                           WriteBuilder();
                     }
                 }
             }
```

And then "WriteBuilder" just Writes the output, it doesn't do a "WriteLine".